model/src/main/proto/math.proto

```diff
@@ -188,3 +188,105 @@ message MathEquation {
   // The MathExpression representing the left-hand side of the equation, e.g. the '0' in 2x+3y=0.
   MathExpression right_side = 2;
 }
+
+// Represents a list of comparable mathematics operations.
+//
+// 'Comparable' here means that this structure provides a trivial way to compare commutative and
+// associative operations (i.e. by extracting terms from multiple subsequent commutative &
+// associative operations into lists that can be deterministically sorted). This structure is meant
+// to provide a means to compare two expressions without considering associativity or commutativity
+// (though the latter requires the operation lists stored within this structure to be sorted before
+// using standard proto equals checking). Also note that care must be taken when performing equality
+// checking since this structure can contain floating point values that require an epsilon check to
+// approximate equality.
+message ComparableOperationList {
+  // An operation that can be compared in a way that does not change the value based on
+  // commutativity or associativity.
+  message ComparableOperation {
+    // Indicates that this operation (e.g. x) should be treated as negated (e.g. -x).
+    bool is_negated = 1;
+
+    // Indicates that this operation (e.g. x) should be treated as a multiplicative inverse
+    // (e.g. 1/x).
+    bool is_inverted = 2;
+
+    // The supported comparison types.
+    oneof comparison_type {
+      // Indicates that this operation is a commutative accumulation (that is, a list of subsequent
+      // operations of the same type that are commutative, e.g. addition or multiplication).
+      CommutativeAccumulation commutative_accumulation = 3;
+
+      // Indicates that this operation is a non-commutative operation and thus cannot be
+      // accumulated (e.g. exponentiation).
+      NonCommutativeOperation non_commutative_operation = 4;
```

Comment:
isn't there assigned numbers that should be 1 and 2 rather than 3 and 4 and below fields too, as these are parts of `oneOf`?

Reply:
The oneof shares indexes with the broader message since it actually is assigning a message-level field. Since ``is_inverted`` has index 2, the fields defined in the oneof need to start at index 3.